ceph_deploy/hosts/remotes.py

```diff
@@ -133,26 +133,29 @@ def write_keyring(path, key):
     tmp_file.close()
     keyring_dir = os.path.dirname(path)
     if not path_exists(keyring_dir):
-        makedir(keyring_dir)
+        makedir(keyring_dir, uid, gid)
     shutil.move(tmp_file.name, path)
 
 
-def create_mon_path(path):
+def create_mon_path(path, uid=-1, gid=-1):
     """create the mon path if it does not exist"""
     if not os.path.exists(path):
         os.makedirs(path)
+        os.chown(path, uid, gid);
 
 
-def create_done_path(done_path):
+def create_done_path(done_path, uid=-1, gid=-1):
     """create a done file to avoid re-doing the mon deployment"""
     with file(done_path, 'w'):
+        os.chown(done_path, uid, gid);
         pass
 
 
-def create_init_path(init_path):
+def create_init_path(init_path, uid=-1, gid=-1):
     """create the init path if it does not exist"""
     if not os.path.exists(init_path):
         with file(init_path, 'w'):
+            os.chown(init_path, uid, gid);
```

Comment:
Agreed.  It also needs to be moved for `create_done_path` immediately before this.